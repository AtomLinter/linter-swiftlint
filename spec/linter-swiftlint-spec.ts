describe("The swiftlint provider for Linter", () => {
  const { lint } = require("../src/index").provideLinter();

  beforeEach(async () => {
    await atom.packages.activatePackage("linter-swiftlint");
  });

  it("should be in the packages list", () => {
    expect(atom.packages.isPackageLoaded("linter-swiftlint")).toBe(true);
  });

  it("should be an active package", () => {
    expect(atom.packages.isPackageActive("linter-swiftlint")).toBe(true);
  });

  it("finds nothing wrong with a well formed file", async () => {
    const editor = await atom.workspace.open(
      __dirname + "/fixtures/well-formed.swift"
    );
    const messages = await lint(editor);

    expect(messages).toMatchSnapshot();
  });

  it("finds something wrong with a not well formed file", async () => {
    const editor = await atom.workspace.open(
      __dirname + "/fixtures/not-well-formed.swift"
    );
    const messages = await lint(editor);

    expect(messages).toMatchSnapshot();
  });
});
