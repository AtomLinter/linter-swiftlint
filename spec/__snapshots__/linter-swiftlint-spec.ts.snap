// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The swiftlint provider for Linter finds nothing wrong with a well formed file 1`] = `Array []`;

exports[`The swiftlint provider for Linter finds something wrong with a not well formed file 1`] = `
Array [
  Object {
    "excerpt": "Identifier Name Violation: Variable name should be between 3 and 40 characters long: 's' (identifier_name)",
    "location": Object {
      "file": "../fixtures/not-well-formed.swift",
      "position": Array [
        Array [
          2,
          0,
        ],
        Array [
          2,
          3,
        ],
      ],
    },
    "severity": "error",
  },
]
`;
